judoscale/core/metric.py

```diff
@@ -1,45 +1,53 @@
-import datetime
 import logging
 import re
 from dataclasses import dataclass
+from datetime import datetime, timezone
+from typing import Optional, Tuple
 
 logger = logging.getLogger(__name__)
 
 
 @dataclass
 class Metric:
-    datetime: datetime
-    value: float
+    timestamp: float  # Unix timestamp in fractional seconds
+    value: float  # Metric value in milliseconds
```

Comment:
This comment assumes a measurement of "queue_time". In the event of needing to support "queue_depth", `value` would actually represent the number of jobs instead of milliseconds.

```suggestion
    value: float
```

Afterthought: This really doesn't need to be a float. We don't care about sub-millisecond values, and if we need to measure "jobs" as in the scenario I just described, a float value doesn't make much sense.

No need to change this now, but just something to keep in mind.

Reply:
That's a good spot and a fair point.

Updated it straight away as there's no need to kick the can down the road with this. 🙂